modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rel/IgniteSystemViewScan.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.sql.engine.rel;
+
+import static org.apache.ignite.internal.sql.engine.trait.TraitUtils.changeTraits;
+
+import java.util.List;
+import org.apache.calcite.plan.RelOptCluster;
+import org.apache.calcite.plan.RelOptTable;
+import org.apache.calcite.plan.RelTraitSet;
+import org.apache.calcite.rel.RelInput;
+import org.apache.calcite.rel.RelWriter;
+import org.apache.calcite.rel.hint.RelHint;
+import org.apache.calcite.rex.RexNode;
+import org.apache.calcite.util.ImmutableBitSet;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Relational operator that returns the contents of a system view.
+ */
+public class IgniteSystemViewScan extends ProjectableFilterableTableScan implements SourceAwareIgniteRel {
+
+    private final long sourceId;
+
+    /**
+     * Constructor.
+     *
+     * @param cluster Cluster this relational expression belongs to.
+     * @param traitSet Traits of this relational expression.
+     * @param hints Optimizer hints.
+     * @param projections Projection list.
+     * @param condition Filter condition.
+     * @param table Target system view to read data from.
+     * @param reqColumns Participating columns.
+     */
+    public IgniteSystemViewScan(RelOptCluster cluster, RelTraitSet traitSet,
+            List<RelHint> hints,
+            RelOptTable table,
+            @Nullable List<RexNode> projections,
+            @Nullable RexNode condition,
+            @Nullable ImmutableBitSet reqColumns) {
+
+        this(-1, cluster, traitSet, hints, table, projections, condition, reqColumns);
+    }
+
+    /**
+     * Constructor used for deserialization.
+     *
+     * @param input Serialized representation.
+     */
+    public IgniteSystemViewScan(RelInput input) {
+        super(changeTraits(input, IgniteConvention.INSTANCE));
+
+        Object srcIdObj = input.get("sourceId");
+        if (srcIdObj != null) {
+            sourceId = ((Number) srcIdObj).longValue();
+        } else {
+            sourceId = -1;
```

Comment:
so many times used `-1`, let's introduce constant for it

Reply:
May be we can create an issue to fix this for all scan operations?